conandata.yml

```diff
@@ -1,8 +1,14 @@
 sources:
+  "1.18.0":
+    url: "https://gitlab.freedesktop.org/gstreamer/gst-plugins-base/-/archive/1.18.0/gst-plugins-base-1.18.0.tar.bz2"
+    sha256 : "e180cde03eb326771db788da45f9597543035972b028dab54068383d44118db5"
   "1.16.0":
     url: "https://gitlab.freedesktop.org/gstreamer/gst-plugins-base/-/archive/1.16.0/gst-plugins-base-1.16.0.tar.bz2"
     sha256: "e6ab73e4c3b0e6b112159b89661a8f709e9bcecfc2827466bc4d3e939ff9e14e"
 patches:
+  "1.18.0":
+    - patch_file: ""
+      base_path: "source_subfolder"
```

Comment:
```suggestion
```
Please modify the recipe to use the following at the beginning of `def build(self)`:
```
for patch in self.conan_data.get("patches", {}).get(self.version, []):
    tools.patch(**patch)
```